File: bin/src/main/java/be/zlz/kara/bin/domain/Event.java
package be.zlz.kara.bin.domain;

import be.zlz.kara.bin.domain.converter.MapToSmileConverter;
import org.springframework.http.HttpMethod;

import javax.persistence.*;
import java.time.LocalDateTime;
import java.util.Map;

@Entity
public class Event {

    @Id
    private String id;

    private byte[] body;

    @Enumerated(EnumType.STRING)
    private HttpMethod method;

    @Enumerated(EnumType.STRING)
    private Source source;

    private String location;

    @Convert(converter = MapToSmileConverter.class)
    private Map<String, String> metadata;

    @Convert(converter = MapToSmileConverter.class)
    private Map<String, String> additionalData;

    private String contentType;

    private LocalDateTime eventTime;

    private String originIp;

    private long bodySize;

    @ManyToOne
    private Bin bin;

    protected Event() {

    }

    public Event(String id, byte[] body, HttpMethod method, Source source, String location, Map<String, String> metadata, Map<String, String> additionalData, String contentType, LocalDateTime eventTime, String originIp, long bodySize, Bin bin) {
        this.id = id;
        this.body = body;
        this.method = method;
        this.source = source;
        this.location = location;
        this.metadata = metadata;
        this.additionalData = additionalData;
        this.contentType = contentType;
        this.eventTime = eventTime;
        this.originIp = originIp;
        this.bodySize = bodySize;
        this.bin = bin;
    }

    public String getId() {
        return id;
    }

    public byte[] getBody() {
        return body;
    }

    public HttpMethod getMethod() {
        return method;
    }

    public Source getSource() {
        return source;
    }

    public String getLocation() {
        return location;
    }

    public Map<String, String> getMetadata() {
        return metadata;
    }

    public Map<String, String> getAdditionalData() {
        return additionalData;
    }

    public String getContentType() {
        return contentType;
    }

    public LocalDateTime getEventTime() {
        return eventTime;
    }

    public String getOriginIp() {
        return originIp;
    }

    public long getBodySize() {
        return bodySize;
    }

    public Bin getBin() {
        return bin;
    }
}

File: bin/src/main/java/be/zlz/kara/bin/domain/Event.kt
package be.zlz.kara.bin.domain

import be.zlz.kara.bin.domain.converter.MapToSmileConverter
import be.zlz.kara.bin.domain.enums.Source
import org.springframework.http.HttpMethod
import java.time.LocalDateTime
import javax.persistence.*


@Entity
class Event {

    @Id
    var id: String? = null
        private set

    lateinit var body: ByteArray
        private set

    @Enumerated(EnumType.STRING)
    var method: HttpMethod? = null
        private set

    @Enumerated(EnumType.STRING)
    var source: Source? = null
        private set

    var location: String? = null
        private set

    @Convert(converter = MapToSmileConverter::class)
    var metadata: Map<String, String>? = null
        private set

    @Convert(converter = MapToSmileConverter::class)
    @Column(name = "additional_data")
    var additionalData: Map<String, String>? = null
        private set

    @Column(name = "content_type")
    var contentType: String? = null
        private set

    @Column(name = "event_time")
    var eventTime: LocalDateTime? = null
        private set

    var origin: String? = null
        private set

    @Column(name = "body_size")
    var bodySize: Long = 0
        private set

    @Column(name = "protocol_version")
    var protocolVersion: String? = null
        private set

    @ManyToOne
    var bin: Bin? = null
        private set

    protected constructor()

    constructor(id: String?, body: ByteArray, method: HttpMethod?, source: Source?, location: String?, metadata: Map<String, String>?, additionalData: Map<String, String>?, contentType: String?, eventTime: LocalDateTime?, origin: String?, bodySize: Long, protocolVersion: String?, bin: Bin?) {
        this.id = id
        this.body = body
        this.method = method
        this.source = source
        this.location = location
        this.metadata = metadata
        this.additionalData = additionalData
        this.contentType = contentType
        this.eventTime = eventTime
        this.origin = origin
        this.bodySize = bodySize
        this.protocolVersion = protocolVersion
        this.bin = bin
    }

}